shared/url_generation.py

```diff
@@ -0,0 +1,172 @@
+        blob_object.url_signed = data_tools.build_secure_url(blob_object.url_signed_blob_path, new_offset_in_seconds)
+        blob_object.url_signed_expiry = time.time() + new_offset_in_seconds
+        if type(blob_object) == Image:
+            blob_object.url_signed_thumb = data_tools.build_secure_url(blob_object.url_signed_thumb_blob_path,
+                                                                       new_offset_in_seconds)
+        session.add(blob_object)
+    except Exception as e:
+        msg = traceback.format_exc()
+        logger.error(msg)
+        log['error']['default_url_regenerate'] = msg
+        return blob_object, log
+    return blob_object, log
+
+
+def get_url_from_connector(connector, params, log):
+    """
+        Gets signed URL from given connector.
+    :param connector:
+    :param params:
+    :param log:
+    :return:
+    """
+    connector.connect()
+    print('params', params)
+    response = connector.fetch_data(params)
+    if response is None or response.get('result') is None:
+        msg = f'Error regenerating URL: {params}. Response: {response}'
+        log['error']['connector_s3_client'] = msg
+        logger.error(msg)
+        return None, log
+    signed_url = response.get('result')
+    return signed_url, log
+
+
+def connection_url_regenerate(session: Session,
+                              blob_object: DiffgramBlobObjectType,
+                              connection_id: int,
+                              bucket_name: int,
+                              new_offset_in_seconds: int,
+                              reference_file: File = None) -> [DiffgramBlobObjectType, dict]:
+    """
+        Regenerates signed url from the given connection ID, bucket and blob path.
+    :param session:
+    :param blob_object:
+    :param connection_id:
+    :param bucket_name:
+    :param new_offset_in_seconds:
+    :return:
+    """
+
+    log = regular_log.default()
+    connection = Connection.get_by_id(session = session, id = connection_id)
+    if connection is None:
+        msg = f'connection id: {connection_id} not found.'
+        log['error']['connection_id'] = msg
+        logger.error(msg)
+        return blob_object, log
+    member = get_member(session = session)
+    params = {
+        'bucket_name': bucket_name,
+        'path': blob_object.url_signed_blob_path if reference_file is None else reference_file.bl,
+        'expiration_offset': new_offset_in_seconds,
+        'action_type': 'get_pre_signed_url',
+        'event_data': {
+            'request_user': member.user_id
+        }
+    }
+
+    if connection.integration_name not in ALLOWED_CONNECTION_SIGNED_URL_PROVIDERS:
+        msg = f'Unsupported connection provider for URL regeneration {connection.id}:{connection.integration_name}'
+        log['error']['unsupported'] = msg
+        logger.error(msg)
+        return blob_object, log
+
+    connection_strategy = ConnectionStrategy(
+        connection_class = S3Connector,
```

Comment:
`connection_class` would be populated based on the connection itself right? maybe need like a look up table for that